### PR TITLE
Minor fixes to async notes

### PR DESCRIPTION
**Pull request by djmitche**
One of these is easy, and the other is removing a nonsense line that I added. If you see how it makes sense, let me know and I'll add it back :)

### CODE REVIEW

**Comment by djmitche** [comment]
Oh, I stand corrected - that was added by @sakex in 281a8e61 (I forgot that this repo squashes PRs).  @sakex, what is "Try making the `sleep` call mutable." demonstrating?

**Comment by sakex** [comment]
> Oh, I stand corrected - that was added by @sakex in 281a8e61 (I forgot that this repo squashes PRs).  @sakex, what is "Try making the `sleep` call mutable." demonstrating?

It becomes much harder to deal with `&mut self` in async trait context. I wanted to illustrate that by offering the students to experiment with this. It's optional and can be removed

**Review by sakex — APPROVED** [review]

**Comment by djmitche** [comment]
Huh, I added it with no trouble -- I just had to make the iteration mutable (`for sleeper in &mut sleepers`) -- should something more complex have been required?

File: src/async/control-flow/join.md [hunk 1]
```diff
@@ -43,9 +43,8 @@ Copy this example into your prepared `src/main.rs` and run it from there.
   cause your program to stall. 
 
 * You can also combine `join_all` with `join!` for instance to join all requests
-  to an http service as well as a database query.
-
-* Try adding a timeout to the future, using `futures::join!`.
+  to an http service as well as a database query. Try adding a
+  `tokio::time::sleep` to the future, using `futures::join!`. This is not a
+  timeout (that requires `select!`, the next chapter), but demonstrates `join!`.
```
Comment (rbehjati) [inline src/async/control-flow/join.md:48]: super nit!

```suggestion
  timeout (that requires `select!`, explained in the next chapter), but demonstrates `join!`.
```

**Review by rbehjati — APPROVED** [review]

**Pull request merged**